substrate/runtime/example/src/lib.rs

```diff
@@ -0,0 +1,385 @@
+/// cursumstances that have happened that users, Dapps and/or chain explorers would find
+/// interested and otherwise difficult to detect.
+#[cfg_attr(feature = "std", derive(Serialize, Deserialize, Debug))]
+#[derive(Encode, Decode, PartialEq, Eq, Clone)]
+pub enum RawEvent<B> {
+	// Just a normal `enum`, here's a dummy event to ensure it compiles.
+	/// Dummy event, just here so there's a generic type that's used.
+	Dummy(B),
+}
+
+// By convention we implement any trait for which a "null implemntation" makes sense
+// for `()`. This is the case for conversion of module `Event` types and hook traits. It
+// is helpful for test code and production configurations where no eventing is necessary
+// or the hook is unused.
+impl<B> From<RawEvent<B>> for () {
+	fn from(_: RawEvent<B>) -> () { () }
+}
+
+decl_storage! {
+	// A macro for the Storage trait, and its implementation, for this module.
+	// This allows for type-safe usage of the Substrate storage database, so you can
+	// keep things around between blocks.
+	trait Store for Module<T: Trait> as Example {
+		// Any storage declarations of the form:
+		//   `pub? Name get(getter_name)? : [required | default]? <type>;`
+		// where `<type>` is either:
+		//   - `Type` (a basic value item); or
+		//   - `map [ KeyType => ValueType ]` (a map item).
+		//
+		// Note that there are two optional modifiers for the storage type declaration.
+		// - `Foo: u32`:
+		//   - `Foo::put(1); Foo::get()` returns `Some(1)`;
+		//   - `Foo::kill(); Foo::get()` returns `None`.
+		// - `Foo: required u32`:
+		//   - `Foo::put(1); Foo::get()` returns `1`;
+		//   - `Foo::kill(); Foo::get()` panics.
+		// - `Foo: default u32`:
+		//   - `Foo::put(1); Foo::get()` returns `1`;
+		//   - `Foo::kill(); Foo::get()` returns `0` (u32::default()).
+		// e.g. Foo: u32;
+		// e.g. pub Bar get(bar): default map [ T::AccountId => Vec<(T::Balance, u64)> ];
+		//
+		// For basic value items, you'll get a type which implements
+		// `runtime_support::StorageValue`. For map items, you'll get a type which
+		// implements `runtime_support::StorageMap`.
+		//
+		// If they have a getter (`get(getter_name)`), then your module will come
+		// equiped with `fn getter_name() -> Type` for basic value items or
+		// `fn getter_name(key: KeyType) -> ValueType` for map items.
+		Dummy get(dummy): T::Balance;
+	}
+}
+
+// The main implementation block for the module. Functions here fall into three broad
+// categories:
+// - Implementations of dispatch functions. The dispatch code generated by the module macro
+// expects each of its functions to be implemented.
+// - Public interface. These are functions that are `pub` and generally fall into inspector
+// functions that do not write to storage and operation functions that do.
+// - Private functions. These are your usual private utilities unavailable to other modules.
+impl<T: Trait> Module<T> {
+	/// Deposit one of this module's events. This function doesn't change.
```

Comment:
Not sure I understand the `This function doesn't change.` part. Can you rephrase?

Reply:
removed as superfluous